src/utils/loose-equal.js

```diff
@@ -16,44 +20,31 @@ function isObject(obj) {
  * Returns boolean true or false
  */
 function looseEqual(a, b) {
-  if (a === b) return true
-  const isObjectA = isObject(a)
-  const isObjectB = isObject(b)
-  if (isObjectA && isObjectB) {
-    try {
-      const isArrayA = isArray(a)
-      const isArrayB = isArray(b)
-      if (isArrayA && isArrayB) {
-        return (
-          a.length === b.length &&
-          a.every((e, i) => {
-            return looseEqual(e, b[i])
-          })
-        )
-      } else if (a instanceof Date && b instanceof Date) {
-        return a.getTime() === b.getTime()
-      } else if (!isArrayA && !isArrayB) {
-        const keysA = keys(a)
-        const keysB = keys(b)
-        return (
-          keysA.length === keysB.length &&
-          keysA.every(key => {
-            return looseEqual(a[key], b[key])
-          })
-        )
-      } else {
-        /* istanbul ignore next */
+  if (a === b) {
+    return true
+  }
+  if (isObject(a) && isObject(b)) {
+    if (keys(a).length !== keys(b).length) {
+      return false
+    }
+
+    // Using for loop over `Object.keys()` here since some class
+    // keys are not handled correctly otherwise
+    for (const key in a) {
+      const aHasKey = a.hasOwnProperty(key)
+      const bHasKey = b.hasOwnProperty(key)
+      if ((aHasKey && !bHasKey) || (!aHasKey && bHasKey) || !looseEqual(a[key], b[key])) {
         return false
```

Comment:
Since this returns false, and the fallback at the end returns false, object equality (loose) will not return `true` reliably when two objects have the same shape (but different key orders).